src/cpp/rtps/xmlparser/XMLElementParser.cpp

```diff
@@ -12,22 +12,160 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 //
+
 #include <cstring>
 #include <regex>
 #include <string>
 #include <unordered_map>
 
 #include <tinyxml2.h>
 
-#include <fastrtps/xmlparser/XMLParserCommon.h>
+#include <fastdds/dds/log/Log.hpp>
+#include <fastrtps/utils/IPLocator.h>
 #include <fastrtps/xmlparser/XMLParser.h>
+#include <fastrtps/xmlparser/XMLParserCommon.h>
 #include <fastrtps/xmlparser/XMLProfileManager.h>
-#include <fastrtps/utils/IPLocator.h>
-#include <fastdds/dds/log/Log.hpp>
 
-using namespace eprosima::fastrtps;
+#include <rtps/xmlparser/XMLParserUtils.hpp>
+#include <utils/SystemInfo.hpp>
+
+template<typename _Enum>
+static bool get_element_enum_value(
+        const char* input,
+        _Enum& output,
+        const char* name,
+        _Enum value)
+{
+    if (0 == strcmp(input, name))
+    {
+        output = value;
+        return true;
+    }
+
+    return false;
+}
+
+template<typename _Enum, typename ... Targs>
+static bool get_element_enum_value(
+        const char* input,
+        _Enum& output,
+        const char* name,
+        _Enum value,
+        Targs... args)
+{
+    if (0 == strcmp(input, name))
+    {
+        output = value;
+        return true;
+    }
+
+    return get_element_enum_value(input, output, args ...);
+}
+
+namespace eprosima {
+namespace fastdds {
+namespace xml {
+namespace detail {
+
+static std::string process_environment(
+        const std::string& input)
+{
+    std::string ret_val = input;
+    std::regex expression("\\$\\{([A-Z0-9_]+)\\}");
```

Comment:
I think it'd be good to document here and in the docs the limitation in naming.

From [IEEE Std 1003.1]:(https://pubs.opengroup.org/onlinepubs/000095399/basedefs/xbd_chap08.html)

Environment variable names used by the utilities in the Shell and Utilities volume of IEEE Std 1003.1-2001 consist solely of uppercase letters, digits, and the '_' (underscore) from the characters defined in [Portable Character Set](https://pubs.opengroup.org/onlinepubs/000095399/basedefs/xbd_chap06.html#tagtcjh_3) and do not begin with a digit.

Once important thing here is that names shall not start with a digit. This is something we aren't validating here

Reply:
712affe